test/runtests.jl

```diff
@@ -355,6 +355,20 @@ using Random # for `randstring`
             end
         end
     end
+
+    @testset "fitsread" begin
+        fname = tempname() * ".fits"
+        a = ones(3,3)
+        try
+            FITS(fname, "w") do f
+                write(f, a)
+            end
+            @test FITSIO.fitsread(fname) == a
+            @test FITSIO.fitsread(fname, 1) == a
+        finally
+            rm(fname, force=true)
+        end
```

Comment:
If you use `mktempdir` you don't need the `try`/`catch` block:
```suggestion
        mktempdir() do dir
            fname = joinpath(dir, "file.fits")
            a = ones(3,3)
            FITS(fname, "w") do f
                write(f, a)
            end
            @test FITSIO.fitsread(fname) == a
            @test FITSIO.fitsread(fname, 1) == a
        end
```
Same thing in the test below

Reply:
Thanks for this idea! I have updated the tests to replace all the `try/catch` blocks with `mktempdir`. Other than this, file closings and cleanups are done through `do`-blocks wherever possible.